Assets/ModScripts/Submodules/LogicalColorCombinations.cs

```diff
@@ -0,0 +1,316 @@
+        { "E", "9", "G", "O", "A", "P", "1", "J" },
+        { "5", "T", "L", "3", "O", "E", "I", "C" },
+        { "K", "6", "D", "R", "F", "0", "X", "Q" },
+        { "Q", "1", "8", "X", "R", "3", "T", "K" },
+        { "U", "Y", "B", "N", "8", "K", "4", "S" },
+        { "Z", "Z", "Z", "Z", "0", "0", "0", "0" },
+        { "0", "0", "0", "0", "Z", "Z", "Z", "Z" },
+    };
+    readonly int[,] logicOperatorTable = new int[11, 14]
+    {
+        { 1, 4, 0, 2, 4, 5, 3, 3, 4, 0, 2, 0, 5, 1 },
+        { 3, 5, 2, 1, 3, 0, 2, 0, 4, 3, 5, 1, 2, 3 },
+        { 2, 0, 4, 3, 3, 4, 1, 2, 1, 5, 2, 3, 0, 5 },
+        { 5, 3, 1, 0, 4, 2, 5, 3, 2, 4, 1, 5, 3, 0 },
+        { 4, 0, 5, 0, 0, 1, 3, 0, 4, 0, 1, 2, 3, 4 },
+        { 0, 5, 2, 3, 1, 4, 5, 5, 1, 2, 4, 2, 0, 5 },
+        { 1, 0, 3, 5, 2, 2, 4, 4, 0, 5, 3, 1, 4, 2 },
+        { 2, 5, 1, 5, 2, 3, 3, 5, 3, 1, 4, 0, 5, 2 },
+        { 3, 1, 4, 2, 5, 0, 4, 1, 5, 5, 0, 2, 1, 4 },
+        { 4, 2, 3, 0, 5, 2, 1, 2, 5, 1, 4, 3, 4, 0 },
+        { 0, 5, 4, 1, 1, 1, 0, 2, 5, 1, 4, 3, 4, 0 }
+    };
+    readonly string[,] pairOrderTable = new string[11, 14]
+    {
+        { "0123", "0213", "0123", "0123", "1302", "1302", "1302", "0123", "1302", "0213", "1203", "1302", "1203", "0312" },
+        { "0312", "0123", "1302", "0312", "0213", "0213", "0312", "1203", "0213", "1203", "2301", "1302", "0123", "2301" },
+        { "1203", "0312", "2301", "0123", "1302", "0123", "0213", "2301", "0213", "0123", "0213", "0312", "0123", "0123" },
+        { "1302", "0213", "0123", "1302", "0213", "0213", "1203", "0213", "0312", "1203", "1203", "0213", "1302", "2301" },
+        { "0312", "0213", "0312", "0123", "0312", "1203", "2301", "0123", "2301", "0312", "0312", "1302", "0312", "0213" },
+        { "2301", "2301", "0312", "0312", "1302", "1203", "0213", "0312", "1302", "0123", "0312", "0123", "1203", "1302" },
+        { "1203", "1203", "2301", "0123", "0213", "1203", "0312", "0123", "1302", "1203", "1203", "0213", "1302", "0213" },
+        { "2301", "0213", "1302", "1203", "0123", "1203", "0213", "0123", "1302", "1302", "1302", "2301", "0312", "0312" },
+        { "1203", "0312", "1203", "2301", "0312", "1203", "2301", "1203", "2301", "0213", "0312", "2301", "0123", "0123" },
+        { "0123", "0213", "0123", "1302", "2301", "1302", "2301", "1203", "1302", "0123", "2301", "0123", "0312", "0213" },
+        { "0123", "0213", "1302", "1302", "0213", "1203", "2301", "1203", "1302", "0123", "2301", "0123", "0312", "0213" }
+    };
+    readonly string[] operatorNames = { "AND", "OR", "XOR", "NAND", "NOR", "XNOR" };
+    readonly string base36 = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+    readonly string[] ArrowDirections = { "Up", "Down", "Left", "Right" };
+    int[] flashes = new int[8];
+    string[] LogABCD = new string[4];
+    int[] ABCD = new int[4];
+    int logicOperator;
+    string pairOrder = "";
+    string[] operationPairs = new string[2];
+    string[] finalPairs = new string[2];
```

Comment:
These fields can all be readonly, along with the solvedLEDs and solvedColors lists